examples/metrics_simple/metrics_ostream.cc

```diff
@@ -27,59 +33,75 @@ namespace
-  std::unique_ptr<metric_sdk::MeterSelector> observable_meter_selector{
-      new metric_sdk::MeterSelector(name, version, schema)};
-  std::unique_ptr<metric_sdk::View> observable_sum_view{
-      new metric_sdk::View{name, "test_description", metric_sdk::AggregationType::kSum}};
+
+  auto observable_instrument_selector = metrics_sdk::InstrumentSelectorFactory::Create(
+      metrics_sdk::InstrumentType::kObservableCounter, observable_counter_name);
+
+  auto observable_meter_selector = metrics_sdk::MeterSelectorFactory::Create(name, version, schema);
+
+  auto observable_sum_view = metrics_sdk::ViewFactory::Create(name, "test_description",
+                                                              metrics_sdk::AggregationType::kSum);
+
   p->AddView(std::move(observable_instrument_selector), std::move(observable_meter_selector),
              std::move(observable_sum_view));
 
   // histogram view
   std::string histogram_name = name + "_histogram";
-  std::unique_ptr<metric_sdk::InstrumentSelector> histogram_instrument_selector{
-      new metric_sdk::InstrumentSelector(metric_sdk::InstrumentType::kHistogram, histogram_name)};
-  std::unique_ptr<metric_sdk::MeterSelector> histogram_meter_selector{
-      new metric_sdk::MeterSelector(name, version, schema)};
-  std::shared_ptr<opentelemetry::sdk::metrics::AggregationConfig> aggregation_config{
-      new opentelemetry::sdk::metrics::HistogramAggregationConfig};
-  static_cast<opentelemetry::sdk::metrics::HistogramAggregationConfig *>(aggregation_config.get())
-      ->boundaries_ = std::vector<double>{0.0,    50.0,   100.0,  250.0,   500.0,  750.0,
-                                          1000.0, 2500.0, 5000.0, 10000.0, 20000.0};
-  std::unique_ptr<metric_sdk::View> histogram_view{new metric_sdk::View{
-      name, "description", metric_sdk::AggregationType::kHistogram, aggregation_config}};
+
+  auto histogram_instrument_selector = metrics_sdk::InstrumentSelectorFactory::Create(
+      metrics_sdk::InstrumentType::kHistogram, histogram_name);
+
+  auto histogram_meter_selector = metrics_sdk::MeterSelectorFactory::Create(name, version, schema);
+
+  auto histogram_aggregation_config = std::unique_ptr<metrics_sdk::HistogramAggregationConfig>(
+      new metrics_sdk::HistogramAggregationConfig);
+
+  histogram_aggregation_config->boundaries_ = std::vector<double>{
+      0.0, 50.0, 100.0, 250.0, 500.0, 750.0, 1000.0, 2500.0, 5000.0, 10000.0, 20000.0};
+
+  std::shared_ptr<metrics_sdk::AggregationConfig> aggregation_config(
+      std::move(histogram_aggregation_config));
+
+  auto histogram_view = metrics_sdk::ViewFactory::Create(
+      name, "description", metrics_sdk::AggregationType::kHistogram, aggregation_config);
+
   p->AddView(std::move(histogram_instrument_selector), std::move(histogram_meter_selector),
              std::move(histogram_view));
+
+  std::shared_ptr<opentelemetry::metrics::MeterProvider> provider(std::move(u_provider));
```

Comment:
These multiple casts from `api:::provider` -> `sdk::provider` -> `api::provider` are bit confusing, but seems we don't have better choice for now.